### PR TITLE
feat: snapshots generated in multi txn has same prev_snapshot_id

### PR DESCRIPTION
**Pull request by SkyFan2002**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

This PR maintains a property: For any snapshot s, if `s.prev` != `null`, then `s.prev` must have been successfully committed, which will be the basis of the new vacuum protocol

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16044)

### CODE REVIEW

**Comment by SkyFan2002** [comment]
replaced by #16044

File: tests/sqllogictests/suites/base/09_fuse_engine/09_0100_insert_multi_table.sql
Comment (dantengsky) [inline tests/sqllogictests/suites/base/09_fuse_engine/09_0100_insert_multi_table.sql:1]: do we need a case that inserting into the same table several times, and check the number of snapshots is as expected?

Comment (SkyFan2002) [reply tests/sqllogictests/suites/base/09_fuse_engine/09_0100_insert_multi_table.sql:1]: Yes, this case is covered in [06_0002_stream_txn_consume.test](https://github.com/datafuselabs/databend/pull/16044/files#diff-a211ffd0518e43111423f4b7ec18596e85509b6e0654ddc8641483c752d9d541)

**Comment by dantengsky** [comment]
some tweaks 

https://github.com/SkyFan2002/databend/pull/11

please take a look.

**Review by dantengsky — APPROVED** [review]

**Pull request merged**